lib/instrumentation/fastify.js

```diff
@@ -88,4 +88,12 @@ module.exports = function initialize(agent, fastify, moduleName, shim) {
       return fastifyForWrapping
     }
   })
+
+  propagateOwnProperties(fastify, wrappedExport)
```

Comment:
I get what this is doing but I don't see how it is working.  The versioned tests install fastify v3 but the agent has a dep of v2.15.2 so it never uses what's in the versioned tests.  If I try to do `npm i fastify@latest` in root of agent and add these tests, it's not wrapping the exports on `fastify` and `default`

```
/*
 * Copyright 2021 New Relic Corporation. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

'use strict'

const tap = require('tap')

const helper = require('../../lib/agent_helper')

tap.test('Fastify Instrumentation', (t) => {
  t.autoend()

  let agent = null

  t.beforeEach(() => {
    agent = helper.instrumentMockedAgent({
      feature_flag: {
        fastify_instrumentation: true
      }
    })
  })

  t.afterEach(() => {
    delete require.cache[require.resolve('fastify')]
    helper.unloadAgent(agent)
  })

  /**
   * Fastify v3 has '.fastify' and '.default' properties attached to the exported
   * 'fastify' function.
   */
  t.test('Should propagate fastify properties when instrumented', (t) => {
    const fastifyExport = require('fastify')
    const original = fastifyExport.__NR_original

    for (const [key, value] of Object.entries(original)) {
      t.equal(fastifyExport[key], value)
    }

    t.end()
  })

  t.test('Should propagate fastify properties when instrumented fastify ', (t) => {
    const { fastify: namedExport } = require('fastify')
    const original = namedExport.__NR_original

    for (const [key, value] of Object.entries(original)) {
      t.equal(fastifyExport[key], value)
    }

    t.end()
  })

  t.test('Should propagate fastify properties when instrumented default', (t) => {
    const { default: defaultExport } = require('fastify')
    const original = defaultExport.__NR_original

    for (const [key, value] of Object.entries(original)) {
      t.equal(fastifyExport[key], value)
    }

    t.end()
  })
})
```

`NR_original` doesn't exist on the `fastify` and `default` keys.  Am I misunderstanding something?

Reply:
Ok I may understand why those tests I added are invalid but don't we want tests that use the named export and default export? right now we just use `module.exports = fastify`

Reply:
Correct. If you read the release note on this PR it explicitly states it is not bothering with instrumenting `.fastify` or `.default`. That is not the *current* known issue for this particular customer. Reading through the GitHub issue may provide more context.

Given we are going to potentially change how we do fastify instrumentation, or at the very least will be deep diving into instrumentation, I did not want to waste time fixing an issue in a way that may be thrown away. At this point, we do not know if we need to instrument `.fastify` or `.deafult` for any use cases. We'll want to dig into that with the fastify effort.

As far as when this will get tested, it is an odd setup but the dev who set this up has the integration tests also ran by the versioned tests. I've added this to the file that pulls those in so the test is in fact ran by the versioned tests on v3. When we deep-dive, I'd like to at a minimum have those ran as separate file. But really, I'm not sure I see the point in the integration test + hard fastify ref. Stuff to consider.

Reply:
I have confirmed though that the fastify version getting installed in versioned tests does not impact the version in integration. it's always running on v2.  If you do `npm i fastify@latest` at root these versioned tests fail

```
# Subtest: Test Transaction Naming
 14     # Subtest: testing naming for /async-return
 15         not ok 1 - You must register a plugin for handling middlewares, visit fastify.io/docs/latest/Middleware/ for mor    e info.
 16           ---
 17           stack: >
 18             Object.use
 19             (/Users/revans/code/node-newrelic/node_modules/fastify/fastify.js:504:11)
 20
 21             loadMiddleware (/Users/revans/code/node-newrelic/test/integration/fastify/naming.tap.js:20:11)
 22
 23             configureFastifyServer (/Users/revans/code/node-newrelic/test/integration/fastify/naming.tap.js:62:9)
 24
 25             Test.<anonymous> (/Users/revans/code/node-newrelic/test/integration/fastify/naming.tap.js:102:11)
 26           at:
 27             line: 504
 28             column: 11
 29             file: /Users/revans/code/node-newrelic/node_modules/fastify/fastify.js
 30             function: Object.use
 31           type: FastifyError
 32           code: FST_ERR_MISSING_MIDDLEWARE
 33           statusCode: 500
 34           test: "testing naming for /async-return "
 35           source: |2
 36               function use () {
 37                 throw new FST_ERR_MISSING_MIDDLEWARE()
 38             ----------^
 39               }
 40           ...
 41
 42         1..1
 43         # failed 1 test
```

Reply:
That is correct.

Reply:
```
test/versioned/fastify on  propagate-fastify-properties [📦📝] via 🤖 v16.9.0
❯ node integration.tap.js
TAP version 13
# Subtest: Fastify Instrumentation
version 2.15.3
```
```
test/versioned/fastify on  propagate-fastify-properties [📦📝] via 🤖 v16.9.0
❯ cat node_modules/fastify/package.json | jq .version
"3.21.0"
```

Reply:
I've verified it runs fine, against both versions, via the versioned tests. I was not concerned about the existing behavior of running integration tests against a single version.

Reply:
That being said, this seems to be causing grief so I can just refactor all of these tests out of Integration since the overlap is of little value.